File: apps/myjobhunter/backend/app/schemas/jobs/resume_upload_job.py
"""ResumeUploadJob schemas."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class ResumeJobParsedFields(BaseModel):
    """Summary of what the resume parser extracted — stored in result_parsed_fields."""
    summary: str | None = None
    headline: str | None = None
    work_history_count: int = 0
    education_count: int = 0
    skills_count: int = 0


class ResumeUploadJobRead(BaseModel):
    id: uuid.UUID
    profile_id: uuid.UUID
    file_filename: str | None = None
    file_content_type: str | None = None
    file_size_bytes: int | None = None
    status: str
    retry_count: int
    error_message: str | None = None
    result_parsed_fields: ResumeJobParsedFields | None = None
    parser_version: str | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/resume-upload-job/resume-upload-job.ts
export type ResumeUploadJobStatus = "queued" | "processing" | "complete" | "failed" | "cancelled";

export interface ParseProvenanceEntry {
  kind: "work_bullet" | "summary";
  work_index?: number;
  company?: string | null;
  bullet_index?: number;
  text: string;
  unsourced_terms: string[];
}

export interface ParseProvenance {
  checked: boolean;
  flagged: ParseProvenanceEntry[];
}

export interface ResumeJobParsedFields {
  summary: string | null;
  headline: string | null;
  work_history_count: number;
  education_count: number;
  skills_count: number;
  /** Parse-time hallucination-guard verdicts. Absent on jobs parsed
   *  before the guard shipped (2026-07-02); null means unchecked. */
  provenance?: ParseProvenance | null;
}

export interface ResumeUploadJob {
  id: string;
  profile_id: string;
  file_filename: string | null;
  file_content_type: string | null;
  file_size_bytes: number | null;
  status: ResumeUploadJobStatus;
  error_message: string | null;
  result_parsed_fields: ResumeJobParsedFields | null;
  parser_version: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
}
